### PR TITLE
Add custom Unmarshalling to work around bad types

### PR DESCRIPTION
**Pull request by deankarn**
this PR addresses issue where a number is coming through on the username(string) field.

added custom JSON Unmarshalling to handle this issue

### CODE REVIEW

File: admin.go [hunk 1]
```diff
@@ -3,14 +3,42 @@ package intercom
...
+	if err := json.Unmarshal(data, &s); err != nil {
+		return err
+	}
+	a.Type = s.Type
+	a.Name = s.Name
+	a.Email = s.Email
+
+	// only 3 possible combinations nil, string, float64
+	if s.ID != nil {
+		if reflect.TypeOf(s.ID).Kind() == reflect.Float64 {
```
Comment (andreiko) [inline admin.go]: would a type switch work here instead of reflect?

Comment (deankarn) [reply admin.go]: yes, but as far as I know it's the same difference...happy to change if you want

Comment (andreiko) [reply admin.go]: aah, using reflect when avoidable is just bad karma, you know

Comment (deankarn) [reply admin.go]: done

**Review by andreiko — APPROVED** [review]

**Pull request merged**